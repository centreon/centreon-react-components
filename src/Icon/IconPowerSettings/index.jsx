import React from 'react';

import PowerSettings from '@material-ui/icons/PowerSettingsNew';
import MaterialIcon from '../MaterialIcon';

import RoundedInvertedIcon from '../RoundedInvertedIcon';

const RoundedInvertedPowerSettings = RoundedInvertedIcon(PowerSettings);

function IconPowerSettings(props) {
  return (
    <MaterialIcon {...props} aria-label="icon enable disable">
      <RoundedInvertedPowerSettings />
    </MaterialIcon>
  );
}

export default IconPowerSettings;
