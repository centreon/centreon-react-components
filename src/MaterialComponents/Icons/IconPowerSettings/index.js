import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import PowerSettings from '@material-ui/icons/PowerSettingsNew';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    textAlign: 'left',
  },
  icon: {
    color: '#fff',
    cursor: 'pointer',
    backgroundColor: '#009fdf',
    borderRadius: '50%',
    fontSize: 15,
    padding: 3,
  },
  iconLabel: {
    color: '#009fdf',
    fontSize: 12, 
    display: 'inline-block',
    verticalAlign: 'super',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 'bold',
    cursor: 'pointer',
    marginLeft: 5,
  },
  iconNormal: {
    color: '#fff',
    cursor: 'pointer',
    backgroundColor: '#707070',
    borderRadius: '50%',
    fontSize: 15,
    padding: 3,
  },
  iconActive: {
    color: '#fff',
    cursor: 'pointer',
    backgroundColor: '#89b72b',
    borderRadius: '50%',
    fontSize: 15,
    padding: 3,
  },
  iconWrap: {
    display: 'inline-block',
    verticalAlign: 'middle',
  }
}));

function IconPowerSettings({label, active, customStyle, ...rest}) {
  const classes = useStyles();

  return (
    <span {...rest} className={classes.iconWrap}>
      <PowerSettings style={customStyle} className={active ? classes.iconActive : classes.iconNormal} />
      {label && <span className={classes.iconLabel}>{label}</span>}
    </span>
  );
}

export default IconPowerSettings;
