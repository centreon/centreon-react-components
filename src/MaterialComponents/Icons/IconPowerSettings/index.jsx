import React from 'react';
import PowerSettings from '@material-ui/icons/PowerSettingsNew';
import styled from '@emotion/styled';
import MaterialIcon from '../MaterialIcon';

const InvertedPowerSettingsIcon = styled(PowerSettings)(() => ({
  color: '#FFF',
  backgroundColor: '#707070',
  borderRadius: '50%',
  padding: 3,
  fontSize: 15,
}));

function IconPowerSettings(props) {
  return (
    <MaterialIcon {...props}>
      <InvertedPowerSettingsIcon />
    </MaterialIcon>
  );
}

export default IconPowerSettings;
