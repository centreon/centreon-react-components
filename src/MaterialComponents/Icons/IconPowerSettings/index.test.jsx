/* eslint-disable no-undef */

import React from 'react';
import { create } from 'react-test-renderer';
import IconPowerSettings from '.';

describe('IconPowerSettings', () => {
  it('renders', () => {
    const wrapper = create(<IconPowerSettings />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
