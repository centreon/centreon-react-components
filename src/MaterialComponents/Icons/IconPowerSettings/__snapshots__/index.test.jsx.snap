// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconPowerSettings renders 1`] = `
.emotion-1 {
  display: inline-block;
  vertical-align: middle;
  height: 24px;
  cursor: pointer;
  color: #707070;
}

.emotion-0 {
  color: #FFF;
  background-color: #707070;
  border-radius: 50%;
  padding: 3px;
  font-size: 15px;
}

<span
  className="emotion-1"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root emotion-0"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
    />
  </svg>
</span>
`;
